galasa-managers-parent/galasa-managers-testingtools-parent/dev.galasa.selenium.manager/src/main/java/dev/galasa/selenium/internal/Browser.java

```diff
@@ -0,0 +1,157 @@
+package dev.galasa.selenium.internal;
+
+import java.io.IOException;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.edge.EdgeDriver;
+import org.openqa.selenium.edge.EdgeOptions;
+import org.openqa.selenium.firefox.FirefoxDriver;
+import org.openqa.selenium.firefox.FirefoxOptions;
+import org.openqa.selenium.firefox.FirefoxProfile;
+import org.openqa.selenium.ie.InternetExplorerDriver;
+import org.openqa.selenium.ie.InternetExplorerOptions;
+import org.openqa.selenium.firefox.ProfilesIni;
+import org.openqa.selenium.opera.OperaDriver;
+import org.openqa.selenium.opera.OperaOptions;
+import org.openqa.selenium.remote.CapabilityType;
+
+import dev.galasa.framework.spi.ConfigurationPropertyStoreException;
+import dev.galasa.selenium.SeleniumManagerException;
+import dev.galasa.selenium.internal.properties.SeleniumGeckoProfile;
+import dev.galasa.selenium.internal.properties.SeleniumWebDriver;
+import dev.galasa.selenium.internal.properties.SeleniumWebDriverPath;
+
+public enum Browser {
+  FIREFOX, IE, CHROME, EDGE, OPERA;
+
+  final static Log logger = LogFactory.getLog(Browser.class);
+
+  public static WebDriver getWebDriver(String instance) throws IOException, SeleniumManagerException {
+    try {
+      String driver = SeleniumWebDriver.get(instance);
+      switch (getBrowser(driver)) {
+        case FIREFOX:
+          return getGeckoDriver(instance);
+        case CHROME:
+          return getChromeDriver(instance);
+        case OPERA:
```

Comment:
Are we really fussed about OPERA as a browser choice?  would we be better to support a generic rendering engine such as webkit?

Reply:
Sorry didn't see that FIREFOX is driving the gecko driver.  Shouldn't we call it GECKO rather than FIREFOX?

Reply:
Or allow GECKO to be a case so it can be specified

Reply:
Removed OPERA support and renamed instances of FIREFOX to GECKO to better describe the web driver